src/main/java/com/endava/endabank/utils/MailService.java

```diff
@@ -0,0 +1,73 @@
+package com.endava.endabank.utils;
+
+import com.endava.endabank.configuration.MailProperties;
+import com.google.common.annotations.VisibleForTesting;
+import com.sendgrid.Method;
+import com.sendgrid.Request;
+import com.sendgrid.Response;
+import com.sendgrid.SendGrid;
+import com.sendgrid.helpers.mail.Mail;
+import com.sendgrid.helpers.mail.objects.Email;
+import com.sendgrid.helpers.mail.objects.Personalization;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+@Component
+@AllArgsConstructor
+public class MailService {
+
+    private MailProperties mailProperties;
+
+    public Response sendEmail(String emailTo, String name, String link,
+                              String templateId, String asName) throws IOException {
+        if (emailTo == null || name == null || link == null || templateId == null || asName == null) {
+            throw new IllegalArgumentException("All parameters must be not null");
+        }
+        if (emailTo.isEmpty() || name.isEmpty() || link.isEmpty() || templateId.isEmpty() || asName.isEmpty()) {
+            throw new IllegalArgumentException("All parameters must be not empty");
```

Comment:
Hi,

Please always check your spelling, I consider this should be "All parameters must not be empty".